extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -142,18 +200,41 @@ class AmpApesterMedia extends AMP.BaseElement {
    * @return {string}
    * */
   constructUrlFromMedia_(id) {
-    return `${this.rendererBaseUrl_}/interaction/${encodeURIComponent(id)}`;
+    const conditionalChannelId = this.embedOptions_distributionChannelId ?
+      `&channelId=${this.embedOptions_distributionChannelId}` : '';
+    const conditionalInteractionType =
+      `&type=${encodeURIComponent(this.embedOptions_.playlist ? 'playlist' :
+        'editorial')}`;
+    const platform = `&platform=${getPlatform()}`;
+    const cannonicalUrl = document.querySelector('link[rel=\'canonical\']')
+      ? document.querySelector('link[rel=\'canonical\']').getAttribute('href')
+      : document.location.href;
+    return `${this.rendererBaseUrl_}/interaction/${
+      encodeURIComponent(id)}?sdk=amp&canonicalUrl=${
+      encodeURIComponent(cannonicalUrl)}${
+      conditionalChannelId}${
+      conditionalInteractionType}${
+      platform}`;
   }
 
-  /** @param {string} src
-   * @return {!Element}
+  /**
+   * Constructs the media Iframe.
+   * @param src
+   * @param mediaId
+   * @private
    */
-  constructIframe_(src) {
+  constructIframe_(src, media) {
     const iframe = this.element.ownerDocument.createElement('iframe');
     iframe.setAttribute('frameborder', '0');
     iframe.setAttribute('allowtransparency', 'true');
     iframe.setAttribute('scrolling', 'no');
-    iframe.src = src;
+    iframe.setAttribute('data-interaction-id', media.interactionId);
+    iframe.name = /wpcomwidgets/.test(document.location.href) ?
+      document.referrer : document.location.href;
+    const splittedUrl = src.split('?');
+    iframe.originalUrlParams = splittedUrl.length > 1 ? `?${splittedUrl[1]}`
```

Comment:
confused here, your embedded page can't access these values (it can only access `name`), so why setting them?